src/lib/dlt_user.c

```diff
@@ -2008,7 +2022,9 @@ static DltReturnValue dlt_user_log_write_raw_internal(DltContextData *log, const
         }
     }
 
-    memcpy(log->buffer + log->size, data, length);
+    if (data != NULL) {
+        memcpy(log->buffer + log->size, data, length);
+    }
     log->size += length;
```

Comment:
log size should only be increased if copy to buffer has happened. please move into brackets of if-statement.

Reply:
oops.. done.

Reply:
same for args_num?

Reply:
Actually yes. But data is already checked at the beginning of the function so this statement always evaluates to true, I just saw this now. So the if statement could be removed. 

If there would be chances to fail for this statement the log would be also in a inconsistent state as "length" was already copied in the buffer few lines above. So the check for data should always be in the beginning of the function.

Reply:
moved `    log->args_num++;` into the `data != null` block